editor/src/clj/editor/code/data.clj

```diff
@@ -589,18 +589,19 @@
         minimap-width (if visible-minimap? (min (Math/ceil (/ excluding-gutter-width 9.0)) 150.0) 0.0)
         canvas-rect-width (- excluding-gutter-width minimap-width)
         line-count (count lines)
+        canvas-rect (->Rect ^double gutter-width 0.0 canvas-rect-width canvas-height)
         scroll-x (limit-scroll (- canvas-rect-width ^double document-width) scroll-x)
-        scroll-y (limit-scroll (- ^double canvas-height (* line-height line-count)) scroll-y)
+        scroll-tab-x-rect (scroll-tab-x-rect canvas-rect document-width scroll-x)
+        scroll-tab-x-height (if scroll-tab-x-rect (.h scroll-tab-x-rect) 0)
```

Comment:
if horizontal scrol exist use its height, if not use 0 and take it into account when calculate limits